frontend/lib/src/components/core/Block/utils.ts

```diff
@@ -173,3 +174,32 @@ export interface BaseBlockProps {
    */
   disableFullscreenMode?: boolean
 }
+
+/**
+ * Converts a user-specified key to a valid CSS class name.
+ *
+ * @param key - The key to convert.
+ * @returns A valid CSS class name.
+ */
+export function convertKeyToClassName(key: string): string {
```

Comment:
I would change the type of `key` to `string | undefined`. You do the `!key` check in here anyways and then you don't have to write `convertKeyToClassName(key ?? "")` everywhere where you call the function. What do you think?

Reply:
Changed 👍